src/vs/code/node/cliVerboseLogger.ts

```diff
@@ -0,0 +1,46 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import { ChildProcess } from 'child_process';
+import { IDisposable } from 'vs/base/common/lifecycle';
+import { Promises } from 'vs/base/node/pfs';
+import { watchFile } from 'vs/base/node/watcher';
+
+/**
+ * This class provides logging for verbose commands on Mac
+ * as part of #102975
+ */
+export class CliVerboseLogger {
+	public async track(child: ChildProcess, logfile: string): Promise<void> {
```

Comment:
I find this still weird that I pass in `ChildProcess`. In the end it is only being used to have a signal when the child process closes, but why can we not from the outside listen to both the close event or the `track` finishing and deal with it? E.g. `Promise.any(...)`?